src/Motor.Extensions.Hosting.NATS/NATSConsumer.cs

```diff
@@ -29,19 +30,30 @@ public class NATSConsumer<TData> : IMessageConsumer<TData>, IDisposable where TD
 
         public async Task ExecuteAsync(CancellationToken token = default)
         {
-            using var subscription = _client.SubscribeAsync(_options.Topic, _options.Queue, async (sender, args) =>
-                {
-                    await SingleMessageHandling(args.Message, token);
-                });
-
-            while (!token.IsCancellationRequested)
+            if (_subscription == null)
             {
-                await Task.Delay(10, token);
+                return;
```

Comment:
(nitpick) Wouldn't it make sense to log something here as this seems to be a usage error because the user called `ExecuteAsync()` without calling `StartAsync()` first which seems to be required?